pkg/kyma/cmd/test/cmd.go

```diff
@@ -0,0 +1,155 @@
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/kyma-incubator/kymactl/internal"
+	kyma_helm "github.com/kyma-incubator/kymactl/internal/helm"
+	"github.com/kyma-incubator/kymactl/internal/step"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/helm/pkg/helm"
+	"k8s.io/helm/pkg/helm/environment"
+	helm_release "k8s.io/helm/pkg/proto/hapi/release"
+)
+
+type Options struct {
+	step.Factory
+	skip []string
+	verbose bool
+}
+
+var testNamespaces = []string{
+	"kyma-system",
+	"istio-system",
+	"knative-serving",
+	"kyma-integration",
+}
+
+var testReleases = map[string]bool{
+	"core": true,
+	"monitoring": false,
+	"logging": false,
+	"istio": true,
+	"knative": true,
+	"application-connector": true,
+}
+
+//NewCmd creates a new install command
+func NewCmd() *cobra.Command {
+	opts := &Options{}
+	cmd := &cobra.Command{
+		Use:   "test",
+		Short: "test kyma installation",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringArrayVar(&opts.skip, "skip", []string{}, "Skip tests for these releases")
+	cmd.Flags().BoolVar(&opts.NonInteractive, "non-interactive", false, "Do not use spinners")
+	cmd.Flags().BoolVarP(&opts.verbose, "verbose", "v", false, "Print additional output")
+	return cmd
+}
+
+func (opts *Options) Run() error {
+	helmConfig := &environment.EnvSettings{TillerConnectionTimeout: 300}
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", clientcmd.RecommendedHomeFile)
+	if err != nil {
+		return err
+	}
+
+	helmClient, err := kyma_helm.New(helmConfig, kubeConfig)
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range testNamespaces {
+		err := opts.cleanHelmTestPods(ns)
+		if err != nil {
+			return err
+		}
+	}
+
+	for release := range testReleases {
+		s := opts.NewStep(
+			fmt.Sprintf("Testing release %s", release),
+		)
+		s.Start()
+
+		if opts.skipRelease(helmClient, release) {
+			s.Successf("Skipping release %s", release)
+			continue
+		}
+
+		msg, success, err := opts.testRelease(helmClient, s, release)
+		s.Stop(success)
+		if msg != "" && (!success || opts.verbose) {
+			fmt.Println("--- Log ---")
+			fmt.Print(msg)
+			fmt.Println("---")
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (opts *Options) skipRelease(client helm.Interface, release string) bool {
+	for _, skipped := range opts.skip {
+		if skipped == release {
+			return true
+		}
+	}
+
+	if !testReleases[release] {
```

Comment:
Somehow the testing failed for me for the "knative" release. It is not installed in my case, the same as monitoring but monitoring got skipped. Please re-check.

Reply:
````
Cleaning up helm test pods in namespace kyma-system ✅
Cleaning up helm test pods in namespace istio-system ✅
Cleaning up helm test pods in namespace knative-serving ✅
Cleaning up helm test pods in namespace kyma-integration ✅
Testing release core ✅
Skipping release monitoring ✅
Skipping release logging ✅
Testing release istio ✅
Testing release knative ❌
Error: rpc error: code = Unknown desc = release: "knative" not found
rpc error: code = Unknown desc = release: "knative" not found
```

Reply:
You probably have kyma installed from release 0.5 (knative is not there). I'm thinking about reading list of modules (helm releases) instead of having them hardcoded.

Reply:
Now, instead of having hardcoded list of components tool is reading them from installation CR.
This should fix the problem.

Reply:
looks good